homeassistant/components/plex/sensor.py

```diff
@@ -8,32 +8,47 @@
 from homeassistant.helpers.entity import Entity
 from homeassistant.util import Throttle
 
-from .const import DOMAIN as PLEX_DOMAIN, SERVERS
+from .const import CONF_SERVER_IDENTIFIER, DOMAIN as PLEX_DOMAIN, SERVERS
 
-DEFAULT_NAME = "Plex"
 _LOGGER = logging.getLogger(__name__)
 
 MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=1)
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Plex sensor."""
-    if discovery_info is None:
-        return
+    """Set up the Plex sensor platform.
 
-    plexserver = list(hass.data[PLEX_DOMAIN][SERVERS].values())[0]
-    add_entities([PlexSensor(plexserver)], True)
+    Deprecated.
+    """
+    return
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Plex sensor from a config entry."""
+
+    def add_entities(devices, update_before_add=False):
+        """Sync version of async add devices."""
+        hass.add_job(async_add_entities, devices, update_before_add)
+
+    hass.async_add_executor_job(_setup_platform, hass, config_entry, add_entities)
+
+
+def _setup_platform(hass, config_entry, add_entities):
```

Comment:
There are no other side effects than adding the entity in this function. We can remove it and inline into `async_setup_entry`.